### PR TITLE
ensure we have new workflows datastream before we index item

### LINKED ISSUE #72: Initializing a workflow does not reindex the object
Adding a new workflow to the object either via Blue Button or direct service /gem call ...

To reproduce, use dor-services gem to do this: 

  i = Dor::Item.find('druid:aa111bb2222') 
  i.initialize_workflow('abcWF') 

Using a legit workflow. Goto Argo for that druid and notice that the workflow abcWF is not present in the History > Workflows section on the page. 

Click Reindex, and notice that the History > Workflows section is now correct.  Adding a new workflow should send an object update message to the DOR index queue.

### PR DESCRIPTION
**Pull request by drh-stanford**
This fixes #72. It requires `dor-services`. It will *not* work until PR https://github.com/sul-dlss/dor-services/pull/112 is merged.

### CODE REVIEW

**Comment by atz** [comment]
Looks like one test got broken as a side-effect: `./spec/controllers/items_controller_spec.rb:379`

**Comment by drh-stanford** [comment]
i've changed my strategy on this fix and i'm going with an earlier solution. the content refresh works, but there's other cached data that i'm not seeing in `dor-services` wrt workflows. the `content` method seems to break the `ActiveFedora` model in that we're not using the built-in `content=` method to load the new workflows XML content (although the patched `dor-services` does indeed refresh it when `content(true)` is called.

**Comment by jmartin-sul** [comment]
tests for this branch pass on my laptop, and travis seems happy again.  code looks reasonable.  ok to merge?  ping @atz since he pointed out the test breakage initially.

File: app/controllers/items_controller.rb [hunk 1]
```diff
@@ -609,7 +609,12 @@ def add_workflow
       render :status => 500, :text => "#{params[:wf]} already exists!"
       return
     end
-    @object.initialize_workflow(params[:wf])
+    @object.create_workflow(params[:wf])
+
+    # sync up the workflows datastream with workflow service and force a reindex before redirection
+    reindex Dor::Item.find(params[:id])
```
Comment (mejackreed) [inline app/controllers/items_controller.rb]: Are we able to test these lines?

Comment (drh-stanford) [inline app/controllers/items_controller.rb]: See my comment re: testing here: https://github.com/sul-dlss/argo/pull/113 `create_workflow` is unit tested in `dor-services` but the `reindex` call isn't.

**Pull request merged**